### PR TITLE
Bug 958341 - Present "motivating service" to /account/create calls.

### PR DESCRIPTION
**Pull request by ncalexan**
This is used for metrics and for personalizing the welcome emails.
Since no email gets sent for /account/login, it is not needed there.

### CODE REVIEW

**Comment by ncalexan** [comment]
Lightly tested with local unit tests.  No device testing yet, but I doubt there will be issues 'cuz I caught several with unit tests.

File: src/main/java/org/mozilla/gecko/background/fxa/FxAccountClient20.java [hunk 1]
```diff
@@ -94,13 +97,40 @@ public void handleSuccess(int status, HttpResponse response, ExtendedJSONObject
...
+  protected URI getCreateAccountURI(final boolean getKeys, final String service) throws UnsupportedEncodingException, URISyntaxException {
+    final StringBuilder sb = new StringBuilder(serverURI); // serverURI always has a trailing slash.
+    sb.append("account/create");
+    if (getKeys || service != null) {
+      // Be very careful that query parameters are encoded correctly!
+      sb.append("?");
+      if (getKeys) {
+        sb.append("keys=true");
+      }
+      if (service != null) {
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/background/fxa/FxAccountClient20.java]: Can we just mandate that the service is never null, and simplify this? Even allowing the empty string should be fine, no?

**Comment by rnewman** [comment]
LGTM, but I don't see a test that verifies that the right parameter exists in the final URL. Did I miss something?

File: src/main/java/org/mozilla/gecko/background/fxa/FxAccountClient20.java (hunk 1 above)
Comment (ncalexan) [inline src/main/java/org/mozilla/gecko/background/fxa/FxAccountClient20.java]: Yeah, I'll do this.  Empty strings should be fine, I think.

**Comment by ncalexan** [comment]
I made `service` mandatory, and I added a test.

**Pull request merged**